### PR TITLE
Sponsors configurations

### PR DESCRIPTION
**Pull request by andreaWaldren**
JIRA Ticket: (link) (⭐)
https://webapps.es.vt.edu/jira/browse/LIBTD-2342

What does this Pull Request do? (⭐)
Adds fields to configure sponsors data.

What's the changes? (⭐)
I changed the HomepageTop component to Homepage and added the form fields and viewing fields for the sponsors to this component. Users can edit the alt text, and link for each sponsor, upload a new image for an existing sponsor (note: file name must match exactly what is shown in "source" - including the file extension). Users can add a new sponsor with image, or delete a sponsor. Also edited the SiteSponsors component to not display if array is present but empty. A cypress test is included.

How should this be tested?
Log in to the siteAdmin page. Click on "Hompage config" and then choose edit. Edit existing sponsors data, upload a new image for a sponsor, or add a new sponsor. Click update config. The new sponsor information should appear on the homepage.

Additional Notes:
Branch is LIBTD-2343

Interested parties
@yinlinchen

### CODE REVIEW

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                includeShadowDom: true,
+            })
+            .first()
+            .find("button[type='submit']", { includeShadowDom: true })
+            .click({ force: true });
+
+        cy.get("#content-wrapper > div > div > ul")
+            .find(":nth-child(4) > a")
+            .contains("Homepage Config")
+            .click();
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:30]: add `cy.wait(500);` below this line

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    });
+
+    it("Updates first sponsor URL", () => {
+        cy.get("input[value='edit']")
+            .parent()
+            .click();
+        cy.get("#s0_link")
+            .clear()
+            .type("https://www.lib.vt.edu");
+        cy.contains("Update Config").click();
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:41]: add cy.wait(500); below this line

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        cy.contains("URL: https://www.lib.vt.edu").should("be.visible");
+        cy.wait(1000);
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:43]: remove

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    });
+
+    it("Reverses update", () => {
+        cy.get("input[value='edit']")
+            .parent()
+            .click();
+        cy.get("#s0_link")
+            .clear()
+            .type("https://clir.org/");
+        cy.contains("Update Config").click();
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:53]: add cy.wait(500); below this line

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        cy.contains("URL: https://clir.org/").should("be.visible");
+        cy.wait(1000);
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:55]: remove

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    });
+
+    it("Updates existing sponsor image", () => {
+        cy.get("input[value='edit']")
+            .parent()
+            .click();
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:61]: add cy.wait(500); below this line

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            .trigger("change", { force: true });
+        cy.get("#sponsor0_form > section > div.fileUploadField > button.uploadButton").click({ force: true });
+
+        cy.get(
+            '#sponsor0_form > section > div.fileUploadField > [data-test="upload-message"]'
+        )
+            .should("have.attr", "style", "color: green;")
+            .invoke("text")
+            .should("include", "uploaded successfully");
+        cy.wait(1000);
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:77]: remove

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        cy.get("button[aria-label='Add sponsor']").click();
+        cy.get(
+            "#sponsor2_form > section > div.fileUploadField > input[type=file]"
+        )
+            .eq(0)
+            .attachFile(imgPath)
+            .trigger("change", { force: true });
+        cy.get(
+            "#sponsor2_form > section > div.fileUploadField > button.uploadButton"
+        ).click({ force: true });
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:94]: add cy.wait(500); below this line

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        cy.get("#s2_alt").type("Virginia Tech");
+        cy.get("#s2_link").type("https://lib.vt.edu");
+        cy.contains("Update Config").click();
+        cy.contains("Sponsor 3").should("be.visible");
+        cy.contains(
+            "Source: https://img.cloud.lib.vt.edu/sites/images/iawa/sponsor3.png"
+        ).should("be.visible");
+        cy.contains("Alt text: Virginia Tech").should("be.visible");
+        cy.contains("URL: https://lib.vt.edu").should("be.visible");
+        cy.wait(1000);
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:104]: remove

File: cypress/integration/admin_homepage_sponsors_config.spec.js [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    })
+
+    it("Removes new sponsor", () => {
+        cy.get("input[value='edit']")
+            .parent()
+            .click();
+        cy.get(
+            "#sponsor2_form > section > button"
+        ).click()
+        cy.contains("Update Config").click();
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_sponsors_config.spec.js:114]: add cy.wait(500); below this line

File: cypress/integration/admin_homepage_top_config.spec.js [hunk 11]
```diff
@@ -26,10 +26,10 @@ describe("Update Homepage fields and revert", function() {
 
     cy.get("#content-wrapper > div > div > ul")
       .find(":nth-child(4) > a")
-      .contains("Homepage Top Config")
-      .click();
+      .contains("Homepage Config")
+      .click()
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_top_config.spec.js:30]: ;

File: cypress/integration/admin_homepage_top_config.spec.js [hunk 12]
```diff
@@ -26,10 +26,10 @@ describe("Update Homepage fields and revert", function() {
 
     cy.get("#content-wrapper > div > div > ul")
       .find(":nth-child(4) > a")
-      .contains("Homepage Top Config")
-      .click();
+      .contains("Homepage Config")
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_top_config.spec.js:29]: Why you remove `Top`?

File: cypress/integration/admin_homepage_top_config.spec.js [hunk 13]
```diff
@@ -26,10 +26,10 @@ describe("Update Homepage fields and revert", function() {
...
+      .click()
     cy.wait(500);
-    cy.url().should("include", "/siteAdmin");
+    cy.url().should("include", "/siteAdmin")
```
Comment (yinlinchen) [inline cypress/integration/admin_homepage_top_config.spec.js:32]: ;

**Review by yinlinchen — CHANGES_REQUESTED** [review]

File: cypress/integration/admin_homepage_top_config.spec.js (hunk 12 above)
Comment (andreaWaldren) [reply cypress/integration/admin_homepage_top_config.spec.js:29]: Because I renamed that section to just homepage config in my changes for this PR.

**Review by yinlinchen — APPROVED** [review]
Nicely done! Great work!

**Pull request merged**